packages/fx-core/src/plugins/resource/aad/plugin.ts

```diff
@@ -86,23 +86,23 @@ export class AadAppForTeamsImpl {
     await DialogUtils.progress?.start(ProgressDetail.Starting);
     if (config.objectId) {
       await DialogUtils.progress?.next(ProgressDetail.GetAadApp);
-      config = await AadAppClient.getAadApp(config.objectId, isLocalDebug, config.password);
+      config = await AadAppClient.getAadApp(ctx, isLocalDebug ? Messages.EndLocalDebug.telemetry : Messages.EndProvision.telemetry, config.objectId, isLocalDebug, config.password);
       ctx.logProvider?.info(Messages.getLog(Messages.GetAadAppSuccess));
     } else {
       await DialogUtils.progress?.next(ProgressDetail.ProvisionAadApp);
-      await AadAppClient.createAadApp(config);
+      await AadAppClient.createAadApp(ctx, isLocalDebug ? Messages.EndLocalDebug.telemetry : Messages.EndProvision.telemetry, config);
       config.password = undefined;
       ctx.logProvider?.info(Messages.getLog(Messages.CreateAadAppSuccess));
     }
 
     if (!config.password) {
       await DialogUtils.progress?.next(ProgressDetail.CreateAadAppSecret);
-      await AadAppClient.createAadAppSecret(config);
+      await AadAppClient.createAadAppSecret(ctx, isLocalDebug ? Messages.EndLocalDebug.telemetry : Messages.EndProvision.telemetry, config);
       ctx.logProvider?.info(Messages.getLog(Messages.CreateAadAppPasswordSuccess));
     }
 
     await DialogUtils.progress?.next(ProgressDetail.UpdatePermission);
-    await AadAppClient.updateAadAppPermission(config.objectId as string, permissions);
+    await AadAppClient.updateAadAppPermission(ctx, isLocalDebug ? Messages.EndLocalDebug.telemetry : Messages.EndProvision.telemetry, config.objectId as string, permissions);
```

Comment:
The same expression are in different place, can we use a variable for this?
```
isLocalDebug ? Messages.EndLocalDebug.telemetry : Messages.EndProvision.telemetry
```

Reply:
Fixed. Thank you for pointing that out.